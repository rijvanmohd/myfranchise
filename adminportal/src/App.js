import React, { Suspense } from 'react';
import { SnackbarProvider } from 'notistack';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';

import Login from './components/accounts/Login';
import PrivateRoute from './components/common/PrivateRoute';
import Dashboard from './components/pages/Dashboard';
import { InfoProvider } from "./context/authContext";

function NotFound() {
  return(
    <h2>Page Not Found</h2>
  )
}

function App() {

  return (
    <InfoProvider>
      <SnackbarProvider maxSnack={3}>
        <Router>
            <Switch>
              <Suspense fallback={<NotFound/>}>
                <PrivateRoute exact path="/" component={Dashboard} type="private"/>
                <PrivateRoute exact path="/login" component={Login} type="guest"/>
              </Suspense>
            </Switch>
        </Router>
      </SnackbarProvider>
    </InfoProvider>
  );
}

export default App;
