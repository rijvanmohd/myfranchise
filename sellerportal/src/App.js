import React, { Suspense } from 'react';
import { SnackbarProvider } from 'notistack';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';

import Login from './components/accounts/Login';
import PrivateRoute from './components/common/PrivateRoute';
import Dashboard from './components/pages/Dashboard';
import { InfoProvider } from "./context/authContext";

function App() {

  return (
    <InfoProvider>
      <SnackbarProvider maxSnack={3}>
        <Router>
          <Suspense fallback={<div>Page is Loading...</div>}>
            <Switch>
              <PrivateRoute exact path="/" component={Dashboard} type="private"/>
              <PrivateRoute exact path="/login" component={Login} type="guest"/>
            </Switch>
          </Suspense>
        </Router>
      </SnackbarProvider>
    </InfoProvider>
  );
}

export default App;
