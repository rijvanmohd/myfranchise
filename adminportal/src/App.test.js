import React from 'react'
import { Router } from 'react-router-dom'
import {render, cleanup, fireEvent, screen} from '@testing-library/react'
import { createMemoryHistory } from 'history'
import renderer from 'react-test-renderer';
import App from './App'

afterEach(cleanup)

describe('Take a snapshot', () => {
  afterEach(cleanup)
    it('should take a snapshot', () => {
    const { asFragment } = render(<App />)
    
    expect(asFragment(<App />)).toMatchSnapshot()
   })
});

// test('landing on a bad page', () => {
//   const history = createMemoryHistory()
//   history.push('/some/bad/route')
//   render(
//     <Router history={history}>
//       <App />
//     </Router>
//   )

//   expect(screen.getByText(/Page Not Found/i)).toBeInTheDocument()
// })

