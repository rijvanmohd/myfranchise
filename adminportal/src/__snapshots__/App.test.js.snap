// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Take a snapshot should take a snapshot 1`] = `
<DocumentFragment>
  <main
    class="MuiContainer-root MuiContainer-maxWidthXs"
  >
    <div
      class="makeStyles-paper-1"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-2 MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
          />
        </svg>
      </div>
      <h1
        class="MuiTypography-root MuiTypography-h5"
      >
        Sign in
      </h1>
      <form
        class="makeStyles-form-3"
        novalidate=""
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          data-testid="email"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-focused Mui-focused Mui-required Mui-required"
            data-shrink="true"
            for="email"
            id="email-label"
          >
            Email Address
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              autocomplete="email"
              class="MuiInputBase-input MuiOutlinedInput-input"
              id="email"
              name="email"
              required=""
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-7 PrivateNotchedOutline-legendNotched-8"
              >
                <span>
                  Email Address *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          data-testid="password"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
            data-shrink="false"
            for="password"
            id="password-label"
          >
            Password
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              autocomplete="current-password"
              class="MuiInputBase-input MuiOutlinedInput-input"
              id="password"
              name="password"
              required=""
              type="password"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-7"
              >
                <span>
                  Password *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-4 MuiButton-containedPrimary MuiButton-fullWidth"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Sign In
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </main>
  <h2>
    Page Not Found
  </h2>
</DocumentFragment>
`;
